### ISSUE TITLE
Apple TV 4K

### ISSUE THREAD

**Issue by fun4stuff** [rileytestut/AltStore#1037]
Is there any way to use AltStore with Apple TV 4K?

**Comment by QuarterSwede** [comment]
This is being looked into but the main problem with the Apple TV is that developers/users have no control over when it offloads app data. Ex. You can install the emulator Provenance currently and it works really well but then one day it'll have offloaded all the roms, saves, etc and it's empty like on a fresh install. Not a great experience and it's random. Apple needs to allow sideloading and protect those apps' data for it to be worth it.

**Comment by tristan-k** [comment]
@QuarterSwede Do you have a source for this information. I recently purchased a Apple TV 4k and did so under the impression that it is possible to sideload apps and refreshing them weekly with AltStore.

**Comment by QuarterSwede** [comment]
@tristan-k Apple doesn’t really give a lot of details on this, to developer angst, but you can install Provenance and test it yourself.

**Comment by yodaluca23** [comment]
How to do this? I'm confused...

**Comment by QuarterSwede** [comment]
> How to do this? I'm confused...

https://wiki.provenance-emu.com/installation-and-usage/installing-provenance/sideloading

**Comment by yodaluca23** [comment]
> > How to do this? I'm confused...
> 
> https://wiki.provenance-emu.com/installation-and-usage/installing-provenance/sideloading

Do I need a mac? Is their a way to do this on WIndows? With Sidloadly or something?

**Comment by QuarterSwede** [comment]
> > > How to do this? I'm confused...
> > 
> > 
> > https://wiki.provenance-emu.com/installation-and-usage/installing-provenance/sideloading
> 
> Do I need a mac? Is their a way to do this on WIndows? With Sidloadly or something?

Altstore will sideload it on Windows too.

**Comment by yodaluca23** [comment]
> > Do I need a mac? Is their a way to do this on WIndows? With Sidloadly or something?
> 
> Altstore will sideload it on Windows too.

How?

**Comment by QuarterSwede** [comment]
> > > Do I need a mac? Is their a way to do this on WIndows? With Sidloadly or something?
> > 
> > 
> > Altstore will sideload it on Windows too.
> 
> How?

Follow the guide.

**Comment by yodaluca23** [comment]
> > > > Do I need a mac? Is their a way to do this on WIndows? With Sidloadly or something?
> > > 
> > > 
> > > Altstore will sideload it on Windows too.
> > 
> > 
> > How?
> 
> Follow the guide.

I may be stupid, but I can't find anywhere on that page how to install on Apple TV 4K on Windows...

**Comment by QuarterSwede** [comment]
> > > > > Do I need a mac? Is their a way to do this on WIndows? With Sidloadly or something?
> > > > 
> > > > 
> > > > Altstore will sideload it on Windows too.
> > > 
> > > 
> > > How?
> > 
> > 
> > Follow the guide.
> 
> I may be stupid, but I can't find anywhere on that page how to install on Apple TV 4K on Windows...

Holding the option key on the Mac version shows the “sideload ipa” option and you can pick your target device and ipa. I’d imagine the alt key does that on windows.

**Comment by yodaluca23** [comment]
> > I may be stupid, but I can't find anywhere on that page how to install on Apple TV 4K on Windows...
> 
> 
> 
> Holding the option key on the Mac version shows the “sideload ipa” option and you can pick your target device and ipa. I’d imagine the alt key does that on windows. That may be a beta only option now that I think about it.

Where? In AltServer? Also how do I pair to my Apple TV?

**Comment by QuarterSwede** [comment]
> > > I may be stupid, but I can't find anywhere on that page how to install on Apple TV 4K on Windows...
> > 
> > 
> > Holding the option key on the Mac version shows the “sideload ipa” option and you can pick your target device and ipa. I’d imagine the alt key does that on windows. That may be a beta only option now that I think about it.
> 
> Where? In AltServer? Also how do I pair to my Apple TV?

Yes, AltServer. You don’t pair it with your Apple TV. You’re only using AltServer to deploy the ipa to the Apple TV. You will have to reinstall it manually every 7 days. AltStore is not available for the Apple TV.

**Comment by yodaluca23** [comment]
> > > Holding the option key on the Mac version shows the “sideload ipa” option and you can pick your target device and ipa. I’d imagine the alt key does that on windows. That may be a beta only option now that I think about it.
> > 
> > 
> > Where? In AltServer? Also how do I pair to my Apple TV?
> 
> Yes, AltServer. You don’t pair it with your Apple TV. You’re only using AltServer to deploy the ipa to the Apple TV. You will have to reinstall it manually every 7 days. AltStore is not available for the Apple TV.

Ok, I finally figured out the button for windows, its shift, but I don't see my Apple TV listed in the list...

**Comment by JackSalzman** [comment]
> > > > Holding the option key on the Mac version shows the “sideload ipa” option and you can pick your target device and ipa. I’d imagine the alt key does that on windows. That may be a beta only option now that I think about it.
> > > 
> > > 
> > > Where? In AltServer? Also how do I pair to my Apple TV?
> > 
> > 
> > Yes, AltServer. You don’t pair it with your Apple TV. You’re only using AltServer to deploy the ipa to the Apple TV. You will have to reinstall it manually every 7 days. AltStore is not available for the Apple TV.
> 
> Ok, I finally figured out the button for windows, its shift, but I don't see my Apple TV listed in the list...

you have to pair it in Xcode. If the TV does not have a USB-C port, you have to pair it wirelessly. 

1   |   On the Apple TV, go to settings --> paired remotes and devices --> pair new device
2   |   Open Xcode. in the menu bar, navigate to the "window" tab
3   |   Click devices and simulators
4   |   In the discovered tab area, click on the name of your Apple TV
5   |   Select Pair
6   |   Enter code being displayed on Apple TV
7   |   Wait for device to be available for development
8   |   TV is paired in Xcode and ready to go
9   |   TV shows up in AltStore
X   |   hold option key (shift key on windows) and click on AltStore. you can now see the option to side load .ipa files.

Note: the .ipa file must be an Apple TV app (built for Apple TV). Most iOS apps will not install or work on Apple TV.

**Comment by JackSalzman** [comment]
(no need to thank me)

**Comment by yodaluca23** [comment]
> > > > > Holding the option key on the Mac version shows the “sideload ipa” option and you can pick your target device and ipa. I’d imagine the alt key does that on windows. That may be a beta only option now that I think about it.
> 
> > > > 
> 
> > > > 
> 
> > > > Where? In AltServer? Also how do I pair to my Apple TV?
> 
> > > 
> 
> > > 
> 
> > > Yes, AltServer. You don’t pair it with your Apple TV. You’re only using AltServer to deploy the ipa to the Apple TV. You will have to reinstall it manually every 7 days. AltStore is not available for the Apple TV.
> 
> > 
> 
> > Ok, I finally figured out the button for windows, its shift, but I don't see my Apple TV listed in the list...
> 
> 
> 
> you have to pair it in Xcode. If the TV does not have a USB-C port, you have to pair it wirelessly. 
> 
> 
> 
> 1   |   On the Apple TV, go to settings --> paired remotes and devices --> pair new device
> 
> 2   |   Open Xcode. in the menu bar, navigate to the "window" tab
> 
> 3   |   Click devices and simulators
> 
> 4   |   In the discovered tab area, click on the name of your Apple TV
> 
> 5   |   Select Pair
> 
> 6   |   Enter code being displayed on Apple TV
> 
> 7   |   Wait for device to be available for development
> 
> 8   |   TV is paired in Xcode and ready to go
> 
> 9   |   TV shows up in AltStore
> 
> X   |   hold option key (shift key on windows) and click on AltStore. you can now see the option to side load .ipa files.
> 
> 
> 
> Note: the .ipa file must be an Apple TV app (built for Apple TV). Most iOS apps will not install or work on Apple TV.

But isn't Xcode only for Mac?

**Comment by eirikgrindevoll** [comment]
This confuses me too. There is like some references some places that AltServer can pair? with Apple tv wireless?

**Comment by W9JYD** [comment]
Hey!!! I’m not the OP but seriously, people? You are giving a ***Windows*** user instructions for macOS!!! Can’t you guys read??? The OP is using ***Windows***!!! There’s no “Option” key on any Windows keyboard! So please answer the question appropriately. The OP is running Windows 10. A ***Microsoft*** product. And so am I. 

Because I am having the same problem and this is the first result in Google. 

This is so frustrating to see you people giving incorrect answers to someone who is not using macOS!!!

**Comment by aayushbhatia06** [comment]
Just checking here as I cannot find updated information anywhere. If I side-load an app using AltServer on my Apple TV (using a non developer account) will I have to resign it every 7 days?

**Comment by QuarterSwede** [comment]
Yes. That’s an Apple limitation.

Aaron

**Comment by luckystarwars** [comment]
if it is so easy please any one can make a video from scratch ton explain how you can install this app for nob people for appletv that have not usb -C or A to connect in direct and use wifi
